op-supervisor/supervisor/backend/activation/activation.go

```diff
@@ -0,0 +1,208 @@
+}
+
+type ActivationManager struct {
+	depSet  depset.DependencySet
+	logger  log.Logger
+	emitter event.Emitter
+}
+
+func NewActivationManager(depSet depset.DependencySet, logger log.Logger) *ActivationManager {
+	return &ActivationManager{
+		depSet: depSet,
+		logger: logger,
+	}
+}
+
+func (am *ActivationManager) AttachEmitter(emitter event.Emitter) {
+	am.emitter = emitter
+}
+
+func (am *ActivationManager) IsActive() bool {
+	return am.IsActiveAt(uint64(time.Now().Unix()))
+}
+
+func (am *ActivationManager) IsActiveAt(timestamp uint64) bool {
+	if timestamp == 0 || am.depSet == nil {
+		return false
+	}
+
+	for _, chain := range am.depSet.Chains() {
+		canInitiate, err := am.depSet.CanInitiateAt(chain, timestamp)
+		if err == nil && canInitiate {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (am *ActivationManager) IsActiveForChain(chain eth.ChainID, timestamp uint64) bool {
+	if timestamp == 0 || am.depSet == nil {
+		return false
+	}
+
+	canInitiate, err := am.depSet.CanInitiateAt(chain, timestamp)
+	if err != nil {
+		am.logger.Error("Error checking interop activation", "chain", chain, "timestamp", timestamp, "err", err)
+		return false
+	}
+	return canInitiate
+}
+
+func (am *ActivationManager) ShouldProcessEvent(chain eth.ChainID, block eth.BlockRef) bool {
+	isActive := am.IsActiveForChain(chain, block.Time)
+	if isActive {
+		am.logger.Info("Potential interop activation detected", "chain", chain, "block", block)
+		return true
+	}
+
+	am.logger.Debug("Filtering pre-interop event", "chain", chain, "block", block)
+	return false
+}
+
+func (am *ActivationManager) DetectAndActivateInterop(
+	ctx context.Context,
+	chain eth.ChainID,
+	block eth.BlockRef,
+	getAnchorPoint func(ctx context.Context) (types.DerivedBlockRefPair, error),
+	isInitialized func(id eth.ChainID) bool,
+	initialize func(id eth.ChainID, anchor types.DerivedBlockRefPair),
+) error {
+	if isInitialized(chain) {
+		return nil
+	}
+
+	if !am.IsActiveForChain(chain, block.Time) {
+		return nil
+	}
+
+	am.logger.Info("Interop activation detected, fetching anchor point", "chain", chain, "block", block)
+	anchor, err := getAnchorPoint(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get anchor point at interop activation: %w", err)
+	}
+
+	if err := am.CheckAnchorPointExpiry(anchor); err != nil {
+		return err
+	}
+
+	am.logger.Info("Initializing with anchor point at interop activation",
+		"chain", chain, "derived", anchor.Derived, "source", anchor.Source)
+	initialize(chain, anchor)
+
+	if am.emitter != nil {
+		am.emitter.Emit(superevents.InteropActivatedEvent{
+			ChainID: chain,
+			Anchor:  anchor,
+			Block:   block,
+		})
+	}
+
+	return nil
+}
+
+func (am *ActivationManager) CheckAnchorPointExpiry(anchor types.DerivedBlockRefPair) error {
+	anchorTime := time.Unix(int64(anchor.Derived.Time), 0)
+	window := time.Duration(am.depSet.MessageExpiryWindow()) * time.Second
+
+	if time.Since(anchorTime) > window {
```

Comment:
Bug: This can't be right, otherwise you couldn't sync after `expiry window` seconds after interop activates.